### DISCUSSION TITLE
Working with Firebase

### DISCUSSION THREAD

**Discussion by dadixon** [pointfreeco/swift-composable-architecture#728] [General]
Has anyone had any success getting TCA to work with Firebase?

**Comment by technoplato** [comment]
My friend and I actually just started working on that a couple days ago.

We're using the [Speech Recognition](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/SpeechRecognition/SpeechRecognition/SpeechClient/Live.swift#L14-L79) Example as a jumping off point, using the analogy of streaming data from the Speech Client as a mental model for Realtime Firestore updates.

You can follow along with out progress on this repo: https://github.com/technoplato/FirebaseComposable

Any tips from the guys would be awesome as we're a little unsure of mocking the client

**Comment by dadixon** [comment]
Has there been any forward progress on this repo? It seems to be private now

**Comment by surajbarkale** [comment]
The latest Firebase SDK now has publisher API's as [a community extension](https://github.com/firebase/firebase-ios-sdk/tree/master/FirebaseCombineSwift).  This greatly simplifies integration with TCA as you don't have to deal with callback based code and just sprinkle `eraseToEffect()` in your environment.

**Reply by technoplato** [reply]
> The latest Firebase SDK now has publisher API's as [a community extension](https://github.com/firebase/firebase-ios-sdk/tree/master/FirebaseCombineSwift).  This greatly simplifies integration with TCA as you don't have to deal with callback based code and just sprinkle `eraseToEffect()` in your environment.

Do you have a wiring example with TCA? I'm still working on grokking and am also building a coreML integration so would be awesome to see a demo if I've got one. If not thanks for the heads up
